lib/spack/spack/test/util/environment.py

```diff
@@ -170,5 +174,10 @@ def test_escape_double_quotes_in_shell_modifications():
         assert 'export VAR="$PATH:$ANOTHER_PATH"' in cmds
         assert r'export QUOTED_VAR="\"MY_VAL\""' in cmds
     else:
-        assert "export VAR=$PATH;$ANOTHER_PATH" in cmds
-        assert r'export QUOTED_VAR="MY_VAL"' in cmds
+        is_pwsh = os.environ.get("SPACK_SHELL", None) == "pwsh"
+        if is_pwsh:
+            assert "$Env:VAR=$PATH;$ANOTHER_PATH" in cmds
+            assert r'$Env:QUOTED_VAR="MY_VAL"' in cmds
```

Comment:
If I understand correctly, the CI doesn't actually cover this. IMO it would be worth simulating this test both with and without powershell (e.g. using a context manager that sets `SPACK_SHELL`).

Reply:
Will do!

Reply:
I've added this to the test and pushed.